### ISSUE TITLE
Can you guys add redis-cli ?

### ISSUE THREAD

**Issue by qiulang** [supercharge/redis-github-action#8]
Currently I add redis-tools myself, so my action.yml looks like this.  I think it will be helpful if redis-cli is included in your action

    ...
    - name: Start Redis
      uses: supercharge/redis-github-action@1.4.0
      with:
        redis-version: ${{ matrix.redis-version }}

    - name: Install redis cli
      run: sudo apt-get install -y redis-tools

**Comment by marcuspoehls** [comment]
@qiulang this GitHub Action adds a docker container running Redis. The Redis CLI is probably available inside of the container.

**Comment by qiulang** [comment]
Yes I noticed your action using docker container. But I am sure the github action can not access it because without installing redis-cli myself, my action failed with the exact the error message redis-cli not found.

The problem with installing it myself if that it seems to take a long time for installation to complete (around 20 seconds). If adding that to your action it will take around 30 seconds (on average), while if I just https://github.com/marketplace/actions/actions-setup-redis it takes around 10 seconds.

**Comment by marcuspoehls** [comment]
Ok, looks like your GitHub action needs the Redis CLI. You’re probably using something like this, are you?

```yaml
...
- name: Start Redis
  uses: supercharge/redis-github-action@1.4.0
  with:
    redis-version: ${{ matrix.redis-version }}

- name: Install redis cli
  run: sudo apt-get install -y redis-tools

- name: Run Redis CLI command
  run: redis-cli ping
```

You need to run some Redis setup from the action’s host machine?

I’m curious: what’s missing in [actions-setup-redis](https://github.com/marketplace/actions/actions-setup-redis) that you’re not using it?

**Comment by qiulang** [comment]
Yes I use `redis-cli ping` to make sure redis-server is up because my app needs redis and it happened in my host environment it fails to start redis-server.


I don't get you words "what’s missing in [actions-setup-redis](https://github.com/marketplace/actions/actions-setup-redis) that you’re not using it?"  I just change the 2 steps I listed into the following step.

```
    - name: Start Redis using actions-setup-redis@v1
      uses: shogo82148/actions-setup-redis@v1
      with:
        redis-version: ${{ matrix.redis-version }}
```

**Comment by marcuspoehls** [comment]
Ok, so you want to make sure Redis is up and running before moving on. We can add that to this action. In case Redis doesn't start up properly, we can stop the pipeline. 

I’m hesitant to install the Redis CLI on the host machine from this action. I would like to keep it reduced to the docker container.

**Comment by qiulang** [comment]
I understand. So at first I just added it myself (without opening this issue) But after I saw [actions-setup-redis](https://github.com/marketplace/actions/actions-setup-redis) uses much less time I was wondering how do they do that ?

**Comment by marcuspoehls** [comment]
They download the Redis source code, compile it, cache it and then start Redis. The caching is probably bringing the speed improvements

**Comment by qiulang** [comment]
@marcuspoehls  I know this issue has been updated for a while but recently I built a docker image with redis server running inside, basically like what you did here and found something strange:

Although my code (running by github action) can access the redis server running inside my docker container, the redis-cli I installed in github action can't access it. When run `redis-cli ping`, I got the error `Error: Connection reset by peer`. I didn't figure out why.

Not sure if you met this error before but this may make my issue a bit harder.
